.github/actions/build-platform-docker/action.yml

```diff
@@ -98,6 +111,31 @@ runs:
       shell: bash
       run: echo "result=$(realpath --relative-to="${PWD}" ${{ inputs.distball }})" >> $GITHUB_OUTPUT
 
+    - name: Import Secrets
+      id: secrets
+      uses: hashicorp/vault-action@v3.0.0
+      if: |
+        inputs.secret_vault_address != ''
+        && inputs.secret_vault_roleId != ''
+        && inputs.secret_vault_secretId != ''
+        && inputs.preview-env == 'true'
+      with:
+        url: ${{ inputs.secret_vault_address }}
+        method: approle
+        roleId: ${{ inputs.secret_vault_roleId }}
+        secretId: ${{ inputs.secret_vault_secretId }}
+        secrets: |
+          secret/data/github.com/organizations/camunda NEXUS_USR;
+          secret/data/github.com/organizations/camunda NEXUS_PSW;
+
+    - name: Login to Harbor docker registry
+      uses: docker/login-action@v3
+      if: inputs.preview-env == 'true'
+      with:
+        registry: registry.camunda.cloud
+        username: ${{ steps.secrets.outputs.NEXUS_USR }}
+        password: ${{ steps.secrets.outputs.NEXUS_PSW }}
```

Comment:
This logic can be moved outside of this file since the `build-platform-docker` action does not need to know about secrets and authentication. See for example how `operate-ci-build-reusable` does the Vault secret handling and login to Harbor before calling `build-platform-docker`. The same approach should be used for `preview-env-build` where you can get secrets and login to Harbor before calling this action - this is cleaner separation of concerns.

The new inputs above are also unnecessary then.

Reply:
I created a new reusable workflow called `zeebe-ci-build-reusable.yml` and another one for Optimize as well